data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCrac.java

```diff
@@ -265,14 +211,21 @@ public State getPreventiveState() {
     @Override
     public Set<State> getStates(Instant instant) {
         return states.values().stream()
-            .filter(state -> state.getInstant().getId().equals(instant.getId()))
+            .filter(state -> state.getInstant().equals(instant))
             .collect(Collectors.toSet());
     }
 
     @Override
     public State getState(Contingency contingency, Instant instant) {
+        if (contingency == null) {
+            if (instant.equals(Instant.PREVENTIVE)) {
+                return getPreventiveState();
+            } else {
+                return null;
+            }
+        }
         return states.values().stream()
-            .filter(state -> state.getContingency().isPresent() && state.getInstant().getId().equals(instant.getId()))
+            .filter(state -> state.getContingency().isPresent() && state.getInstant().equals(instant))
```

Comment:
why call state.getContingency().isPresent() before state.getInstant().equals(instant)?
it may be more efficient to merge the two filters and order the conditions starting with the most constraining one (state.getInstant().equals(instant) &&  state.getContingency().isPresent() && state.getContingency().get().getId().equals(contingency.getId()) )

Reply:
I kept two filters for readability but if it impacts performance I can merge it into one